### PR TITLE
Add support for inspect.signature

### LINKED ISSUE #137: Doubled methods signature can't be inspected
Trying to `inspect.signature` a double method raises the following (shortened traceback):

```
  File "my_code/base.py", line 279, in func
    inspect.signature(f).parameters
  File "/usr/local/Cellar/python/3.6.5_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/inspect.py", line 3037, in signature
    return Signature.from_callable(obj, follow_wrapped=follow_wrapped)
  File "/usr/local/Cellar/python/3.6.5_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/inspect.py", line 2787, in from_callable
    follow_wrapper_chains=follow_wrapped)
  File "/usr/local/Cellar/python/3.6.5_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/inspect.py", line 2266, in _signature_from_callable
    skip_bound_arg=skip_bound_arg)
  File "/usr/local/Cellar/python/3.6.5_1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/inspect.py", line 2090, in _signature_from_builtin
    raise ValueError("no signature found for builtin {!r}".format(func))
ValueError: no signature found for builtin <doubles.proxy_method.ProxyMethod object at 0x115fe8240>
```

Should be fairly easy to fix since the PEP allows signature to be provided via `__wrapped__` or `__signature__`.

Will provide a PR to fix this.

### PR DESCRIPTION
**Pull request by charlax**
Fixes #137.

Happy to remove the autoformatting done by `black`.

### CODE REVIEW

File: doubles/proxy_method.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 
 def double_name(name):
-    return 'double_of_' + name
+    return "double_of_" + name
```
Comment (skudriashev) [inline doubles/proxy_method.py]: Why this change is needed? `"` -> `'`

**Comment by skudriashev** [comment]
Please remove all irrelevant changes. It is hard to review.

File: doubles/proxy_method.py (hunk 1 above)
Comment (skudriashev) [reply doubles/proxy_method.py]: Please don't mix different changes :)

**Comment by charlax** [comment]
Should be good now.

**Comment by skudriashev** [comment]
LGTM. Please squash before merge.

**Comment by charlax** [comment]
I don't have merge access - I don't work at Uber anymore.

**Comment by charlax** [comment]
@skudriashev @toddsifleet would it be possible to issue a release? Thanks!

**Comment by skudriashev** [comment]
@charlax released :)

**Comment by charlax** [comment]
Awesome thanks!
